shared/middleware/helpers/storage_apps.rb

```diff
@@ -45,6 +48,20 @@ def get(channel_id)
     row = @table.where(id: id).exclude(state: 'deleted').first
     raise NotFound, "channel `#{channel_id}` not found" unless row
 
+    # For some apps, if it was created by a signed out user, we don't want anyone
+    # else to be able to access it (for privacy reasons)
+    anonymous_age_restricted_apps = ['applab', 'gamelab', 'weblab']
+    if owner != @storage_id && !storage_id_for_user_id(owner)
+      begin
+        # row[:projectType] isn't set for channels associated with levels (vs. standalone
+        # projects), so we crack open the JSON blob instead
+        project_type = JSON.parse(row[:value])['projectType']
```

Comment:
We could (and perhaps should) make it so that we set the project_type column even for level-based channels, but that has wider ramifications and seems more likely to break something.